### ISSUE TITLE
Phenotype modeling for IMPC

### ISSUE THREAD

**Issue by bryanlaraway** [monarch-initiative/dipper#54]
References #14 .

Have a question regarding the modeling of additional phenotype data in IMPC. I have created the genotype to phenotype associations, but I'm unsure exactly how I should proceed in creating the phenotype node and where to hang the additional phenotype testing data.

For the genotype to phenotype association, I'm currently using the MP term as the phenotype ID, same as we are using for the [NIF view](http://beta.neuinfo.org/mynif/search.php?q=*&t=indexable&nif=nlx_151660-3), although the ID is hidden in the interface. However, I think we don't want to hang the additional phenotype data available from IMPC, such as a phenotype free text description (includes phenotype center, testing procedure, testing parameter, effect size, and p-value), from the phenotype ID, as this would result in a single phenotype associated with multiple genotypes and multiple measurements attached to the phenotype that are associated with multiple genotypes.

Update: Talked with Matt and it sounds appropriate to hang this data to the association between the genotype and the phenotype. He has created GENO_0000612 - 'phenotype description' for the phenotype description free text. Will work on creating additional entries for other phenotype parts as needed.

**Comment by nlwashington** [comment]
@mbrush should we use that new GENO class, or should we just use DC:description?

**Comment by mbrush** [comment]
As Bryan indicated, I think it best to hang these free-text descriptions from the association.  Bryan - after reading your overview above about what is contained in the description, I would advise just using dc:description for now rather than the more specific geno property.

**Comment by bryanlaraway** [comment]
@mbrush One additional item in regards to genotype and phenotype modeling is in regard to the sex of the organism. In the views we created for NIF, the sex was not incorporated into the [MouseGenotypes view](http://beta.neuinfo.org/mynif/search.php?q=*&t=indexable&nif=nlx_151660-2), and the underlying data stopped at the creation of the intrinsic genotype, while the [MousePhenotypes view](http://beta.neuinfo.org/mynif/search.php?q=*&t=indexable&nif=nlx_151660-3) included the sex, and the genotype partonomy was extended to the effective genotype, which in the case of IMPC was the intrinsic genotype + sex. The question then is whether it would be more appropriate to have the sex be part of the intrinsic genotype, allowing for the consideration of how the differences in sex chromosomes apply to the intrinsic genotype, or whether it should stay attached to the effective genotype.

**Comment by nlwashington** [comment]
i wonder if there needs to be some refactoring in GENO to have a concept of the non-sex-chromosome part of the genotype?

**Comment by mbrush** [comment]
Summarizing my thoughts in this treatise on sex in G2P associations. 

Given our use cases for creating and querying data, one viable approach is to treat the sex of the organism as contextualizing information for a G2P association - much like we treat environments. We would simply create a new relation to hang a sex 'qualifier' from the association. Something like:
 
    (1)     Association     [qualifier relation]      male/female
 
Phenotype qualifier relations have not yet been defined or named for other qualifying information, such as environment and developmental stage.  But if we agree this is the best model, we can at least mint an IRI now to use in creating the data, and define the label/semantics of the relation later.  For now we can just call these relations 'sex qualifier',  'environment qualifier', 'stage qualifier' for now - or just use a single has_qualifier relation for all of these qualifiers. 
 
--

Another way to hang the sex from the association is just to treat it as a phenotype (which in reality is all that sex is).  With respect to the structure of the graph, this is no different than the proposal above - its just that the relation between the association and the sex IRI is has_object rather than a qualifier relation.
 
    (2)     genotype has_object  male/female
 
What we **don’t** want to do for the sex 'phenotype' is create the additional direct link from the genotype to the sex as we do for other phenotypes in our graph, i.e. **don’t** say this:
 
    (3)     genotype has_phenotype  male/female
 
This is because we don’t want to have a **_direct_** link between a genotype and sex,  because we want a single identifier for a given genotype, and we would have to duplicate each genotype if we wanted to distinguish male from female ones.
 
Avoidance of a direct genotype-sex link also means that we don’t want a model that defines sex at the genetic level, which we had also discussed, e.g.
 
   (4)     genotype   has_part   male sex chromosome
 
--

Finally, another approach that I think IS valid is linking genotype to sex **_indirectly_** through the organism, which would of course require creating an organism instance/bnode in order to get to the sex:
   
 (5)      genotype     is_genotype_of       organism 
           organism     has_quality             male/female
 

While perhaps the most biologically precise and accurate, this seems unnecessary from the perspective of efficiency/succinctness, and our data requirements. But if we encounter data or use cases that need for this type of modeling, it is easy to implement (as all of the necessary relations live in our import chain already).  And it is not mutually exclusive with (1) and (2).  But we would only want to use this approach if there was utility in instantiating/describing the organism in which phenotypes were inhering. 

In summary, I advocate for (1) or (2) above, but (5) is not mutaully exclusive and could be applied additionally.  (3) and (4) are not practical or efficient given our data and use cases.

**Comment by nlwashington** [comment]
i disagree that sex isn't related to the genotype.  they often have different chromosomes, and there are specific genes that lie on those chromosomes that determine sex.  (except in some insects, perhaps plants, and some reptiles, where there is no Y, but the number of X's or temperature determines phenotypic sex.)  and when there are sex-specific differences on a gene on a sex-chromosome, this is very very important.  (of course, if the gene lies on the autosome, then it is indirectly important.)  but yes, i do think (I) above is a viable option when we want to talk about the context of the phenotype observation.

**Comment by nlwashington** [comment]
and i very much dislike 2/3.  not sure why 4 doesn't work?

**Comment by mbrush** [comment]
I should have been more clear that (3) and (4) are not *wrong*.  I do agree that sex is related to the genotype - in that it is a phenotype resulting from genetic sequence features.   What I was trying to say is that we shouldn't create a direct genotype - sex link in our data, because the genotype instances we identify are intentionally uncommitted as to their sex.  If we wanted a triple directly form a genotype IRI to a sex IRI, we would have to split our genotypes apart into a male and female versions (see below).  It is better that we commit to a sex for a genotype only in the context of a G2P association - as accommodated by (1) or (2). 

Consider the following: if we use patterns such as (4), we would end up with direct triples linking a genotype such as MGI:3584154 Shh<tm1Amc>/Shh<tm1Amc>) to a sex chromosome. 

    MGI:3584154  has_part  female sex chromosome

But we dont want to make this commitment to the sex of MGI:3584154, because we want it to stand for a male or female genotype.  And given that we may have cases where a G2P association refers to a male MGI:3584154  genotype, and others where one refers to a female MGI:3584154 genotype, we cannot link MGI:3584154 directly to a sex without ending up with conflicting triples in our data.  To avoid this we would have to split MGI:3584154 into two genotypes (one that contains female genetic bits and the other that contains male genetic bits).

    MGI:3584154-F  has_phenotype  female
    MGI:3584154-M  has_phenotype  male
  

I'm pretty sure we don't want to do this. This is why I advocate for (1) or (2) - because they require assigning a sex to a genotype only in the context of a G2P association. Apologies if I am overlooking something or not making myself clear . . .  can discuss on next call if so.

**Comment by mbrush** [comment]
Also @nlwashington, don't hate on (2) too strongly (unless it is because you don't think sex is a phenotype, which may be valid). The pattern of (2) looks funny, but recall that has_object is the relation between an Association and a phenotype in our reification model. So if we go with the notion that sex is a phenotype, it is reasonable to describe it analogously to other phenotypes in the context of a given Association if it is specified that the organism was a male or female.  Again, this approach is completely isomorphic with (1)  in terms of the structure of the graph, and should satisfy all query/filtering use cases.  Its just that the name of the Association to sex is has_object instead of some qualifying relationship.

**Comment by nlwashington** [comment]
this requires light refactoring to get in the "sex specific genotype" types and properties.

**Comment by kshefchek** [comment]
This is an old ticket, but we have implemented both 1 and 4.
